plugins/modules/cloud/misc/proxmox_nic.py

```diff
@@ -0,0 +1,350 @@
+    api_user: root@pam
+    api_password: secret
+    api_host: proxmoxhost
+    name: my_vm
+    interface: net0
+    state: absent
+'''
+
+RETURN = '''
+vmid:
+  description: The VM vmid.
+  returned: success
+  type: int
+  sample: 115
+msg:
+  description: A short message
+  returned: always
+  type: str
+  sample: "Nic net0 unchanged on VM with vmid 103"
+'''
+
+try:
+    from proxmoxer import ProxmoxAPI
+    HAS_PROXMOXER = True
+except ImportError:
+    HAS_PROXMOXER = False
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+
+
+def get_vmid(module, proxmox, name):
+    try:
+        vms = [vm['vmid'] for vm in proxmox.cluster.resources.get(type='vm') if vm.get('name') == name]
+    except Exception as e:
+        module.fail_json(msg='Error: %s occurred while retrieving VM with name = %s' % (e, name))
+
+    if not vms:
+        module.fail_json(msg='No VM found with name: %s' % name)
+    elif len(vms) > 1:
+        module.fail_json(msg='Multiple VMs found with name: %s, provide vmid instead' % name)
+
+    return vms[0]
+
+
+def get_vm(proxmox, vmid):
+    return [vm for vm in proxmox.cluster.resources.get(type='vm') if vm['vmid'] == int(vmid)]
+
+
+def update_nic(module, proxmox, vmid, interface, model, **kwargs):
+    vm = get_vm(proxmox, vmid)
+
+    try:
+        vminfo = proxmox.nodes(vm[0]['node']).qemu(vmid).config.get()
+    except Exception as e:
+        module.fail_json(msg='Getting information for VM with vmid = %s failed with exception: %s' % (vmid, e))
+
+    if interface in vminfo:
+        # Convert the current config to a dictionary
+        config = vminfo[interface].split(',')
+        config.sort()
+
+        config_current = {}
+
+        for i in config:
+            kv = i.split('=')
+            try:
+                config_current[kv[0]] = kv[1]
+            except IndexError:
+                config_current[kv[0]] = ''
+
+        # determine the current model nic and mac-address
+        models = ['e1000', 'e1000-82540em', 'e1000-82544gc', 'e1000-82545em', 'i82551', 'i82557b',
+                  'i82559er', 'ne2k_isa', 'ne2k_pci', 'pcnet', 'rtl8139', 'virtio', 'vmxnet3']
+        current_model = set(models) & set(config_current.keys())
+        current_model = current_model.pop()
+        current_mac = config_current[current_model]
+
+        # build nic config string
+        config_provided = "{0}={1}".format(model, current_mac)
+    else:
+        config_provided = model
+
+    if kwargs['mac']:
+        config_provided = "{0}={1}".format(model, kwargs['mac'])
+
+    if kwargs['bridge']:
+        config_provided += ",bridge={0}".format(kwargs['bridge'])
+
+    if kwargs['firewall']:
+        config_provided += ",firewall=1"
+
+    if kwargs['link_down']:
+        config_provided += ',link_down=1'
+
+    if kwargs['mtu']:
+        if model == 'virtio':
+            config_provided += ",mtu={0}".format(kwargs['mtu'])
+        else:
+            module.fail_json(
```

Comment:
I like it, I've tested this, proxmox will allow the setting but will then warn when starting the VM that it is ignored because the model is not virtio. The module will now print a warning and ignore the setting instead of failing.

Reply:
Great! Thanks for testing and changing this!